crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -216,28 +267,36 @@ impl Object {
 
     /// The Base64 encoded bcs serialization of the object's content.
     async fn bcs(&self) -> Result<Option<Base64>> {
-        if let Some(stored) = &self.stored {
-            Ok(Some(Base64::from(&stored.serialized_object)))
-        } else {
-            let bytes = bcs::to_bytes(&self.native)
-                .map_err(|e| {
-                    Error::Internal(format!(
-                        "Failed to serialize object at {}: {e}",
-                        self.address,
-                    ))
-                })
-                .extend()?;
-
-            Ok(Some(Base64::from(&bytes)))
-        }
+        use ObjectKind as K;
+        Ok(match &self.kind {
+            K::WrappedOrDeleted(_) => None,
+            K::OutsideAvailableRange => None,
+            K::Live(_, stored) => Some(Base64::from(&stored.serialized_object)),
+            K::Historical(_, stored) => stored.serialized_object.as_ref().map(Base64::from),
```

Comment:
Do we know why the historical form's stored object has a nullable serialized object? At first I thought it was to handle wrapped or deleted objects, but I see that they got their own case.  If there's a reason for it, can you document it here, otherwise, we should go back and make the two tables consistent (as a post-launch follow-up).

Reply:
WrappedOrDeleted objects do not have a serialized object, thus the serialized object column in the db must be nullable